### PR TITLE
Add lint workflow

### PR DESCRIPTION
**Pull request by owenkim**
## What
Add a lint workflow for pull requests for baseline static analysis

### CODE REVIEW

File: .github/workflows/qa-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          mkdir -p ~/Library/MobileDevice/Provisioning\ Profiles
+          cp $PP_PATH ~/Library/MobileDevice/Provisioning\ Profiles
+      - name: Build app
+        run: |
+          ls ~/Library/MobileDevice/Provisioning\ Profiles
+          cd Sample
+          mkdir -p ios_build
+          swift build
+          xcodebuild -scheme SampleForageSDK -destination 'generic/platform=iOS' CONFIGURATION_BUILD_DIR=./ios_build build DEVELOPMENT_TEAM=3D54TBBRF3 -allowProvisioningUpdates CODE_SIGNING_ALLOWED=NO
+          ls
```
Comment (devinmorgan) [inline .github/workflows/qa-tests.yaml:44]: Clarifying question: does this job run a QA-Tests suite? I followed that it builds the SampleForageSDK but wasn't able to discern whether a test suite gets run. Thanks for any clarity!

**Review by devinmorgan — APPROVED** [review]
Not entirely sure how this PR causes the QA Tests to fail? 🤔 That aside, this linting check is super useful - thanks for taking the time improve our dev experience!

File: .github/workflows/qa-tests.yaml (hunk 1 above)
Comment (dleis612) [reply .github/workflows/qa-tests.yaml:44]: It will be doing that once it's done and working :) battling with iOS and github actions is not an easy task.

Comment (owenkim) [reply .github/workflows/qa-tests.yaml]: I'm moving the QA test work to a separate PR so I will merge this to start the linting step.

**Pull request merged**